File: Frequency/5LongestPalindromicSubstring.cpp
/**
 Solution:
 Expand Around Center. Observe that a palindrome mirrors around its center.
*/
class Solution {
public:
    string longestPalindrome(string s) {
        if(s.size()==0) return "";
        string rev = "";
        for(int i=0; i<s.size(); i++){
            rev = helper(s, i, i).size()>rev.size() ? helper(s, i, i) : rev;
            rev = helper(s, i, i+1).size()>rev.size() ? helper(s, i, i+1) : rev;
        }
        return rev;
    }
    string helper(string s, int i, int j){
        while(i>=0 && j<s.size() && s[i]==s[j]){
            i--;
            j++;
        }
        return s.substr(i+1, j-i-1);
    }
};

File: BinaryTree/5longestPalindromicSubstring.py
#from the center check whether or not is a palindrome

class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        def check(s,l,r):
            while l >= 0 and r < len(s) and s[l] == s[r]:
                l -= 1
                r += 1
            return s[l+1:r]
        
        res = ''
        for i in range(len(s)):
            temp = check(s,i,i)
            if len(temp) > len(res):
                res = temp
            temp = check(s,i,i+1)
            if len(temp) > len(res):
                res = temp
        return res